src/tpm2/crypto/openssl/TpmToOsslHash.h

```diff
@@ -75,25 +75,9 @@
 #    endif           // libtpms added end
 #       undef ALG_SM3_256
 #       define ALG_SM3_256  ALG_NO
-#   elif OPENSSL_VERSION_NUMBER >= 0x10200000L
-#       include <openssl/sm3.h>
 #   else
-// OpenSSL 1.1.1 keeps smX.h headers in the include/crypto directory,
-// and they do not get installed as part of the libssl package
-#       define SM3_LBLOCK      (64/4)
-
-# error Check support for this version of SM3 in OpenSSL (libtpms)
-typedef struct SM3state_st {
-    unsigned int A, B, C, D, E, F, G, H;
-    unsigned int Nl, Nh;
-    unsigned int data[SM3_LBLOCK];
-    unsigned int num;
-} SM3_CTX;
-
-int sm3_init(SM3_CTX *c);
-int sm3_update(SM3_CTX *c, const void *data, size_t len);
-int sm3_final(unsigned char *md, SM3_CTX *c);
-#   endif // OpenSSL < 1.2
+    #include "Sm3Helper_fp.h"
+#endif
 #endif // ALG_SM3_256
```

Comment:
Why do you remove it from here? I'd rather not move this here to keep the differences to the reference code at a minimum.